### ISSUE TITLE
tikv recover will take 60m when cancel injection if  setting blockchaos delay to 60m

### ISSUE THREAD

**Issue by Lily2025** [chaos-mesh/chaos-mesh#3672]
## Bug Report

**What version of Kubernetes are you using?**

**What version of Chaos Mesh are you using?**

**What did you do? / Minimal Reproducible Example**

**What did you expect to see?**
    -   name: "one_tikv_io_delay"
        faultType: io_delay(volumeName:tikv,delay:60m)
        selector: "tikv(random)"
        warmUpTime: 10m
        faultDuration: 120m
        recoveryTime: 10m

**What did you see instead?**
![image](https://user-images.githubusercontent.com/84712107/191928354-f30334d6-6715-434a-8c2a-42a5fe04e182.png)

**Output of chaosctl**

**Comment by STRRL** [comment]
It seems we could resolve this issue by updating the delay to `0` before we call the `del_injection`, cc @YangKeao.

**Comment by YangKeao** [comment]
> It seems we could resolve this issue by updating the delay to `0` before we call the `del_injection`, cc @YangKeao.

No. It's an engineering problem. The root cause is that for an io request, the `send_time` of it is calculated at very first, so the last request during the injection will still be sent after 60min, and it blocks the tikv-server process.

Recalculating the `send_time` for every requests when the injection has changed can fix this problem, but the code structure could be a little complicated :thinking:.


As adding 60min latency is used as freeze, but not real latency, I propose to use `ioctl FIFREEZE` to implement a new freeze action for `BlockChaos` (it's a blockchaos because it'll need to select a block device.) It would be much simpler, and is already supported by all linux kernels.
